Firestore/core/src/firebase/firestore/remote/grpc_connection.mm

```diff
@@ -52,9 +54,28 @@
   return view.data() ? std::string{view.data(), view.size()} : std::string{};
 }
 
+std::string LoadCertificate(const Path& path) {
+  std::ifstream certificate_file{path.native_value()};
+  HARD_ASSERT(certificate_file.good(),
+              StringFormat("Unable to open root certificates at file path %s",
+                           path.ToUtf8String())
+                  .c_str());
+
+  std::stringstream buffer;
+  buffer << certificate_file.rdbuf();
+  return buffer.str();
+}
+
+std::shared_ptr<grpc::ChannelCredentials> CreateSslCredentials(
+    const Path& certificate_path) {
+  grpc::SslCredentialsOptions options;
+  options.pem_root_certs = LoadCertificate(certificate_path);
```

Comment:
It seems like means we're going to load this certificate every time we create a channel. Is that how this worked in the Objective-C client we're replacing?

Reply:
No, gRPC Objective-C client [stores the certificate in a static variable](https://github.com/grpc/grpc/blob/master/src/objective-c/GRPCClient/private/GRPCHost.m#L155-L170).

I'm totally fine with doing it that way. My reasoning was this: channel is created infrequently. It's created once for every new `GrpcConnection` object, whose lifetime is tied to `FSTDatastore` (so IIUC, it will only be recreated if the whole Firestore instance is recreated). It may also be recreated if it gets into `GRPC_CHANNEL_SHUTDOWN` state. I can't find any docs on the state; from digging through the code, it seems like it's very rare and seems to be mostly in case of gRPC bugs. On the other hand, certificate is approximately 260 KB to hold in memory -- not huge, but not tiny. All in all, it's a tradeoff. What do you think? I'm fine either way.